src/main.ts

```diff
@@ -174,10 +174,10 @@ export default class ObsidianGit extends Plugin {
     }
 
     migrateSettings() {
-        if(this.settings.mergeOnPull) {
-            this.settings.syncMethod = 'merge'
-            this.settings.mergeOnPull = undefined
-            this.saveSettings()
+        if (this.settings.mergeOnPull != undefined) {
```

Comment:
We have to set `syncMethod` on `rebase` too, don't we?

Reply:
I don't _believe_ so given that `rebase` is the default, but I'm :+1: on making this setting explicit so this looks reasonable to me.

Reply:
That all being said -- if we're going the explicit route, we probably need to make sure this ends up being set even if `this.settings.mergeOnPull` isn't set, no?  Otherwise, I'm pretty sure this particular setting will still possibly be `undefined` for users that generated their settings before this PR.

Reply:
> I don't believe so given that rebase is the default,

No, because the default of `mergeOnPull` is/was `true`

> ends up being set even if this.settings.mergeOnPull isn't set, no?

https://github.com/denolehov/obsidian-git/blob/master/src/main.ts#L182 sets values from the default config for undefined keys in the loaded setting. So this shouldn't be a problem.

Reply:
Oh, neat; I had totally missed that!